src/Sparse/Matrix.jl

```diff
@@ -952,6 +952,172 @@ function Base.insert!(A::SMat{T}, i::Integer, B::SRow{T}) where T
   return A
 end
 
+################################################################################
+#
+#  delete/empty
+#
+################################################################################
+
+@doc Markdown.doc"""
+  delete_row!(A::SMat{T}, i::Int) -> SMat{T}
+
+Deletes $i$-th row of $A$ in place.
+"""
+function delete_row!(A::SMat{T}, i::Int) where T
+  non_zeros = length(A[i].pos)
+  deleteat!(A.rows, i)
+  A.r-=1
+  A.nnz-=non_zeros
+  return A
+end
+
+@doc Markdown.doc"""
+  delete_rows!(A::SMat{T}, I, sorted=true)
+
+Deletes rows in set of indices $I$ of $A$ in place.
+"""
+function delete_rows!(A::SMat{T}, I, sorted=true) where T #elements in I need to be ascending
+  if !sorted
+      sort(I)
+  end
+  for i in length(I):-1:1
+      delete_row!(A, I[i])
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+  delete_zero_rows!(A::SMat{T}, s=1)
+
+Deletes zero rows of $A$ in place.
+"""
+function delete_zero_rows!(A::SMat{T}, s=1) where T #where s denotes the first row where we wanna start
+  for i=A.r:-1:s
+      if A[i].pos == []
```

Comment:
```suggestion
      if length(A[i].pos) == 0
```
Or `isempty(A[i].pos)`.